### PR TITLE
Allow multiple objectives in the optimizer config

### PR DESCRIPTION
**Pull request by motus**
* [x] Update JSON schema to use `"optimizer_target": {"score": "min"}` format
* [x] Update base Optimizer class to use the new config format (throw `NotImplementedError` if > 1 target)
* [x] Modify all unit tests to use the new format

Part of #692

Merge after #723

### CODE REVIEW

File: mlos_bench/mlos_bench/config/schemas/cli/common-defs-subschemas.json [hunk 1]
```diff
@@ -31,6 +30,14 @@
             "description": "Config ID to use (or resume) for the experiment.",
             "type": "integer",
             "minimum": 0
+        },
+        "optimization_target": {
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/config/schemas/cli/common-defs-subschemas.json:34]:
```suggestion
        "optimization_targets": {
```
?

File: mlos_bench/mlos_bench/config/schemas/optimizers/optimizer-schema.json [hunk 2]
```diff
@@ -13,14 +13,7 @@
             "type": "object",
             "properties": {
                 "optimization_target": {
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/config/schemas/optimizers/optimizer-schema.json:15]:
```suggestion
                "optimization_targets": {
```

File: mlos_bench/mlos_bench/optimizers/base_optimizer.py [hunk 3]
```diff
@@ -78,8 +77,15 @@ def __init__(self,
         self._start_with_defaults: bool = bool(
             strtobool(str(self._config.pop('start_with_defaults', True))))
         self._max_iter = int(self._config.pop('max_suggestions', 100))
-        self._opt_target = str(self._config.pop('optimization_target', 'score'))
-        self._opt_sign = {"min": 1, "max": -1}[self._config.pop('optimization_direction', 'min')]
+
+        opt_targets: Dict[str, str] = self._config.pop('optimization_target', {'score': 'min'})
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/optimizers/base_optimizer.py:81]:
```suggestion
        opt_targets: Dict[str, str] = self._config.pop('optimization_targets', {'score': 'min'})
```

For consistency and make the config a little more self documenting.

**Review by bpkroth — CHANGES_REQUESTED** [review]
Just a slight naming change, otherwise LGTM

File: mlos_bench/mlos_bench/tests/config/schemas/optimizers/test-cases/bad/invalid/bad_optimization_target.jsonc
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/config/schemas/optimizers/test-cases/bad/invalid/bad_optimization_target.jsonc:1]: Do we have any tests checking that the old schema is no longer recognized?

Comment (motus) [reply mlos_bench/mlos_bench/tests/config/schemas/optimizers/test-cases/bad/invalid/bad_optimization_target.jsonc:1]: good point, let me add one

**Review by bpkroth — APPROVED** [review]
LGTM!

**Pull request merged**